tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,149 @@
+"""
+Check the auto-restart feature of containers
+"""
+import time
+import pytest
+import logging
+
+from common.utilities import wait_until
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_STOP_THRESHOLD_SECS = 30
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+def get_process_id(duthost, container_name, process):
+    """
+    @summary: Get the pid of process in the specified container by analyzing
+              the command output of "ps -ax"
+    @return: first value is a boolean. True indicates the processs is running
+             second value is the process id. Returning -1 if the process is not found
+    """
+    running_status = False
+    process_id = -1
+    ps_out = duthost.shell("docker exec {} ps -ax".format(container_name))
+    for line in ps_out["stdout_lines"]:
+        process_info = line.split()
+        if "/usr/bin/supervisor-proc-exit-listener" not in process_info \
+            and str(process_info).find(process) != -1:
+            running_status = True
+            process_id = int(process_info[0])
+            break 
+
+    return running_status, process_id
+ 
+def kill_process(duthost, container_name, process):
+    """
+    @summary: Kill a process in the specified container
+
+    """
+    running_status, process_id = get_process_id(duthost, container_name, process)
+    if running_status:
+        duthost.shell("docker exec {} kill -SIGTERM {}".format(container_name, process_id)) 
+
+    running_status, process_id = get_process_id(duthost, container_name, process)
+    if running_status:
+        assert False, "Failed to stop {} process before test.".format(process)
```

Comment:
We are currently experiencing selective issues with `assert` in pytest, where it doesn't behave properly. We have added our own `pytest_assert()` function, which you can use via `from common.helpers.assertions import pytest_assert`, or you can use `pytest.fail()` directly (`pytest_assert()` is a wrapper around `pytest.fail()`). Please replace all `asserts` with one of these.

Reply:
Will update!